### PR TITLE
Fix a bug involving return_initial_states=True and outputs

### PR DESCRIPTION
**Pull request by rizar**
`return_initial_states=True` did not work when not all of the outputs were before all the states.

@dmitriy-serdyuk , this should be an easy PR to review

@pbrakel , this should bring us one step closer towards using Blocks master

### CODE REVIEW

File: blocks/bricks/recurrent/base.py [hunk 1]
```diff
@@ -227,10 +227,11 @@ def scan_function(*args):
...
             if return_initial_states:
                 # Undo Subtensor
-                for i in range(len(states_given)):
-                    assert isinstance(result[i].owner.op,
-                                      tensor.subtensor.Subtensor)
-                    result[i] = result[i].owner.inputs[0]
+                for i, info in enumerate(outputs_info):
+                    if info is not None:
+                        assert isinstance(result[i].owner.op,
+                                        tensor.subtensor.Subtensor)
```
Comment (dmitriy-serdyuk) [inline blocks/bricks/recurrent/base.py]: Indent is wrong here.

File: blocks/bricks/recurrent/base.py [hunk 2]
```diff
@@ -227,10 +227,11 @@ def scan_function(*args):
             result = pack(result)
             if return_initial_states:
                 # Undo Subtensor
-                for i in range(len(states_given)):
-                    assert isinstance(result[i].owner.op,
-                                      tensor.subtensor.Subtensor)
-                    result[i] = result[i].owner.inputs[0]
+                for i, info in enumerate(outputs_info):
+                    if info is not None:
```
Comment (dmitriy-serdyuk) [inline blocks/bricks/recurrent/base.py]: This relies on line 216, right? Can you add a comment that None corresponds to an output and shouldn't be included?

Comment (rizar) [inline blocks/bricks/recurrent/base.py]: The convention that `None` corresponds to an output is not Blocks-specific. It is a part of `theano.scan` signature.

**Comment by rizar** [comment]
A fixed the syntax error you pointed out and also a few others in the same file.

File: blocks/bricks/recurrent/base.py (hunk 2 above)
Comment (dmitriy-serdyuk) [inline blocks/bricks/recurrent/base.py]: Right, agreed.

**Comment by dmitriy-serdyuk** [comment]
LGTM.

**Pull request merged**